File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/chat/ServerVersionChatCommand.java
package net.bestia.zoneserver.chat;

import net.bestia.model.domain.Account.Companion.UserLevel;
import net.bestia.messages.MessageApi;
import net.bestia.model.domain.Account;
import net.bestia.zoneserver.config.StaticConfig;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Objects;

/**
 * Sends the server version to the client.
 * 
 * @author Thomas Felix
 *
 */
@Component
public class ServerVersionChatCommand extends BaseChatCommand {

	private final StaticConfig config;

	@Autowired
	public ServerVersionChatCommand(MessageApi akkaApi, StaticConfig config) {
		super(akkaApi);

		this.config = Objects.requireNonNull(config);
	}

	@Override
	public boolean isCommand(String text) {
		return text.startsWith("/net/bestia/server ") || text.equals("/net/bestia/server");
	}

	@Override
	public UserLevel requiredUserLevel() {
		return UserLevel.USER;
	}

	@Override
	public void executeCommand(Account account, String text) {
		final String replyText = String.format("Bestia Behemoth Server (%s)", config.getServerVersion());
		sendSystemMessage(account.getId(), replyText);
	}

	@Override
	protected String getHelpText() {
		return "Usage: /server";
	}

}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/chat/ServerVersionChatCommand.kt
package net.bestia.zoneserver.chat

import net.bestia.model.account.Account
import net.bestia.model.account.AccountType
import net.bestia.zoneserver.actor.routing.MessageApi
import org.springframework.boot.info.BuildProperties
import org.springframework.stereotype.Component
import java.text.SimpleDateFormat
import java.util.*

/**
 * Sends the server version to the client.
 *
 * @author Thomas Felix
 */
@Component
internal class ServerVersionChatCommand(
    msgApi: MessageApi,
    private val buildProperties: BuildProperties
) : BaseChatCommand(msgApi) {

  override val helpText: String
    get() = "Usage: /serverinfo"

  override fun isCommand(text: String): Boolean {
    return text.startsWith("/serverinfo ") || text == "/serverinfo"
  }

  override fun requiredUserLevel(): AccountType {
    return AccountType.USER
  }

  override fun executeCommand(account: Account, text: String) {
    val dateStr = SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(Date.from(buildProperties.time))
    val replyText = "Bestia Behemoth Server: v.${buildProperties.version} ($dateStr)"
    sendSystemMessage(account.id, replyText)
  }
}
